source/slang/slang-ir-wgsl-legalize.cpp

```diff
@@ -145,11 +145,10 @@ struct LegalizeWGSLEntryPointContext
                     structType,
                     mapOldFieldToNewField,
                     semanticInfoToRemove);
+                // Validate/rearange all semantics which overlap in our flat struct.
+                fixFieldSemanticsOfFlatStruct(flattenedStruct);
```

Comment:
If this function is going to be called even when it is not flattened, the name of the function may need to change to match